R/check-default-branch.R

```diff
@@ -0,0 +1,34 @@
+#' Ensure that default GitHub branch is not "master"
+#'
+#' The `$info$git$branch` value is taken by default from GitHub as long as
+#' "DESC" has a remote URL. It is only taken from local git if not remote
+#' GitHub URL can be identified.`
+#'
+#' @param checks A 'pkgcheck' object with full \pkg{pkgstats} summary and
+#' \pkg{goodpractice} results.
+#' @return TRUE if default GitHub branch is "master"; otherwise FALSE
+#' @noRd
+pkgchk_branch_is_master <- function (checks) {
+
+    ret <- FALSE
+    if (length (checks$info$git) > 0L) {
+        ret <- checks$info$git$branch == "master"
+    }
+
+    return (ret)
+}
+
+output_pkgchk_branch_is_master <- function (checks) {
+
+    out <- list (
+        check_pass = !checks$checks$branch_is_master,
+        summary = "",
+        print = ""
+    )
+
+    if (!out$check_pass) {
+        out$summary <- "Default GitHub branch of 'master' is not acceptable."
```

Comment:
@maelle Can you please think about whether this phrase might be improved? thank you

Reply:
can we add hints, such as a link to the dev guide one this is policy?

Anyhow I noticed but liked how blunt it was :sweat_smile: 

I found `"Package contains unacceptable 'License' entries: ["` so "unacceptable" is an acceptable word in results?

Alternative

```
"Default GitHub branch of 'master' is no longer an acceptable standard."
```

Reply:
I do like that, but it's also a bit long. An unwritten part of the effort here is to ensure all check results are as concise as possible. Any unnecessarily long check results tend to distract focus, which I fear that might do.